File: javascript/raindrops/raindrops.js
let DROPS = {
    3: 'Pling',
    5: 'Plang',
    7: 'Plong',
};

export default class Raindrops {
    convert(drops) {
        let getSound = (a, k) => drops % k == 0 ? a + DROPS[k] : a;
        return Object.keys(DROPS).reduce(getSound, '') || String(drops);
    }
};

File: c/raindrops/raindrops.c
#include "raindrops.h"
#include <stdio.h>

void convert(char * const buffer,
             const int num) {
  size_t len = sprintf(buffer, "%s%s%s",
                       num % 3 == 0 ? "Pling" : "",
                       num % 5 == 0 ? "Plang" : "",
                       num % 7 == 0 ? "Plong" : "");

  if (len <= 0) {
    sprintf(buffer, "%d", num);
  }
}
